src/lightning_app/core/work.py

```diff
@@ -374,6 +380,7 @@ def _default_setattr(self, name: str, value: Any) -> None:
                 value = deepcopy(value)
                 value.component_name = self.name
                 self._state.add(name)
+                self._drives.append(value)
```

Comment:
This isn't the cleanest as you need to validate the drive gets removed when the value is overridden by something else.

Reply:
good call! removed.